modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/TransactionStateResolver.java

```diff
@@ -218,18 +222,17 @@ private void resolveTxStateFromTxCoordinator(
 
         if (coordinator == null) {
             // This means the coordinator node have either left the cluster or restarted.
+            markAbandoned(txId);
```

Comment:
This is not necessary because we will resolve the write intent despite the fact that the state ABANDONED or is not a final one.

Reply:
We will, but we still should put the tx in this state because it's not correct not to do it. For example:
- RO transaction tx0 resolves write intent belonging to the transaction tx1 and marks it as abandoned and starts the recovery;
- after that RW transaction tx2 meets the lock belonging to tx1, sees that it's abandoned recently and doesn't start the recovery, minimizing redundant network messages.

Reply:
I think we cannot avoid the write intent resolution procedure. It does not depend on the transaction state.
Hence, the recovery happens each time we step into a write intent.

Reply:
The given example still shows the need of transition to ABANDONED.